test/xbgpu/test_engine.py

```diff
@@ -470,6 +470,12 @@ def prom_get(name: str) -> float:
         assert prom_get("output_b_samples_total") == n_beam_heaps_sent * heap_samples
         assert saturated_low[i] <= prom_get("output_b_clipped_samples_total") <= saturated_high[i]
 
+        # Check that sensor updates match Prometheus updates
```

Comment:
Not to be a pain, but you're not checking updates plural here, are you? You're checking the last update in the list of updates.
* Nothing wrong with that, it clearly makes sense.
* I'd request you update/tweak the comment to reflect that, please.